pkgs/tools/typesetting/tex/nix/run-latex.sh

```diff
@@ -61,10 +66,7 @@ if test -n "$copySources"; then
 fi
 
 
-echo "PASS 1..."
 runLaTeX
-echo
-
 
 for auxFile in $(find . -name "*.aux"); do
```

Comment:
Would it be possible to detect section references in the aux file?

Reply:
Are you suggesting we check if the `.aux` file contains writes to the `toc` table and only have the additional pass if there are writes?
This could work, but I suspect that the issue might arise with more than just the `toc` table, and having somewhat complex logic to parse `.aux` files and determine if certain tables need additional passes seems error-prone and not worth the optimization.

Furthermore, such an optimization would only be useful when the document both: doesn't have a table of contents, and does not write anything to the table of contents (does not use `\section` and friends) which seems like an **extremely** small use case.

Reply:
If understand your suggestion correctly, then I think it is not worth optimizing